### ISSUE TITLE
[Bug]: Blinking `custom/script` module on continuous script failure with unchanged output

### ISSUE THREAD

**Issue by Equaraphanus** [polybar/polybar#2636]
### Checklist

- [X] I have read the appropriate section in the [contributing guidelines](https://github.com/polybar/polybar/blob/master/CONTRIBUTING.md)
- [X] I believe this issue is a problem with polybar itself and not a misconfiguration on my part
- [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that may have already reported this problem
- [X] I have checked the [known issues](https://github.com/polybar/polybar/wiki/Known-Issues) page for this problem.
- [X] I have followed the [debugging guide](https://github.com/polybar/polybar/wiki/Debugging-your-Config) to narrow down the problem to a minimal config.

### Steps to reproduce

1. Put the minimal config into `blinking_error.ini`;
2. Run `polybar -c blinking_error.ini`;
3. Witness the blinking `error` text.

### Minimal config

```dosini
[bar/test]
modules-left = test-error

[module/test-error]
type = custom/script
exec = "echo error && false"
interval = 1
format = <label>
label = %output%
```

### Polybar log

```text
notice: Parsing config file: blinking_error.ini
warn: No fonts specified, using fallback font "fixed"
notice: Loaded font "fixed" (name=DejaVu Sans, offset=0, file=/usr/share/fonts/TTF/DejaVuSans.ttf)
notice: Received signal(2): Interrupt
notice: Termination signal received, shutting down...
```

### Expected behavior

As long as the output is not _actually_ empty, the module should not disappear, instead it should stay as it is, displaying the output (`error` in this case) with all appropriate formatting applied (if provided), regardless of the command's exit status and if its output has changed since last time or not.

### Actual behavior

What actually happens is this:
![the `error` text is blinking](https://user-images.githubusercontent.com/53881077/157756028-8e8c8924-e9d1-4ade-8961-daa4e4c5b7c0.gif)

Every other script run the output gets reset to `""` and the module disappears completely only to reappear again on the next run.

### Window Manager and Version

bspwm 0.9.10

### Linux Distribution

Arch Linux

### Polybar version

```text
polybar 3.6.1

Features: +alsa +curl +i3 +mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -O3 -DNDEBUG -Wall -Wextra -Wpedantic -Wsuggest-override
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -Wall -Wextra -Wpedantic -Wsuggest-override  -Wall -Wextra -Wpedantic -Wsuggest-override
```

### Additional Context / Screenshots

After some time of `m_log.warn()` debugging, I came across this piece of code, located (as of 880fbc24) in [`script_runner.cpp`, line 130](https://github.com/polybar/polybar/blob/880fbc242db18e46b1cf9c4e66f863cfd8103e88/src/adapters/script_runner.cpp#L130):

    if (!changed && m_exit_status != 0) {
      clear_output();
    }

Basically it forcefully resets the output to `""` when the script prints the same thing again while exiting with non-zero code.

I can't think of any case when this behavior could be useful, especially after the formatting for script failure was introduced in #2596.

If this behavior is required for some reason, it would be nice to have an ability to opt-out of it with some per-module config setting.

**Comment by indev29** [comment]
Hi! Thank you for reporting the issue. 

It's indeed a known "feature" that wasn't "fixed" in #2596 in order to retain the behavior of already existing configurations. Additional option may be a solution to this, although I'm not sure if i like it. 

Personally, I'd rather change the behavior. @patrick96, what do you think?

**Comment by patrick96** [comment]
From what I can tell, this behavior was introduced over 5 years ago in 5773b08cf8709a2b8642f45b5e2f5560df332e22.

I don't know the original intention behind this and I can't find a use-case for this behavior (even less now that there is `format-fail`).

I am generally wary of changing existing behavior because there are always people out there relying on it (even if they don't necessarily know it). But in this case, I think we can get away with introducing a small breaking change and point people towards `format-fail` in the changelog.

If anyone wants to work on this right away, let me know and I'll assign this to you. I'll also add this to the 3.7.0 milestone, so this will definitely get done for that release.

**Comment by z0xyz** [comment]
Facing the exact same issue
